schema.sql

```diff
@@ -37,30 +43,25 @@ CREATE TABLE purchase (
   total DECIMAL NOT NULL
 );
 
+DROP TABLE IF EXISTS delivery_person;
+CREATE TABLE delivery_person (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR( 255 )
+);
+
 
 DROP TABLE IF EXISTS delivery;
 CREATE TABLE delivery (
   id SERIAL PRIMARY KEY,
   status VARCHAR ( 255 ) NOT NULL,
   delivery_address INTEGER REFERENCES address ( id ) NOT NULL,
-  delivery_person VARCHAR ( 255 ) NOT NULL,
+  delivery_person INTEGER REFERENCES delivery_person ( id ) NOT NULL,
   purchase_id INTEGER REFERENCES purchase ( id ) NOT NULL,
   price DECIMAL NOT NULL,
```

Comment:
Do you have `price` existing for non-delivery orders? Unless you are auto-adding delivery fees at this step this could also be a foreign key that references the original price.